src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -799,9 +799,24 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
                     }
                 };
                 params.target = id;
-                params.id = `${currentNodeIdRef.current}${currentHandleIdRef.current}${id}`;
-                params.data.id = `${currentNodeIdRef.current}${currentHandleIdRef.current}${id}`;
+                let relationshipAmount = 0;
+                while (
+                    elements.some(
+                        (element) =>
+                            element.source &&
+                            element.source === currentNodeIdRef.current &&
+                            element.id ===
+                                `${currentNodeIdRef.current}${currentHandleIdRef.current}${relationshipAmount};${versionClassBase}`
+                    )
+                ) {
+                    relationshipAmount++;
+                }
```

Comment:
This relationship count code, or something similar, is in a couple of places in the file now. I think we should extract it into a function that can be reused.